django/contrib/postgres/fields/jsonb.py

```diff
@@ -107,7 +107,8 @@ def as_sql(self, compiler, connection):
             previous = previous.lhs
         lhs, params = compiler.compile(previous)
         if len(key_transforms) > 1:
-            return "(%s %s %%s)" % (lhs, self.nested_operator), [key_transforms] + params
+            key_transforms_str = "{%s}" % ",".join(key_transforms)
```

Comment:
I think this exposes us to SQL injection vulnerabilities? I think we'll want to keep passing `key_transforms` as params.

Not sure how possible that is but I'd be great if we only added the `{...}` if necessary because for the same reasons described in https://code.djangoproject.com/ticket/29482#comment:1

Reply:
hmm... you are right, let me try to figure out how to do that without exposing SQL injection.

Actually we can convert the list into tuple, so the params can be serialize to string and avoid the error described in original ticket. But we still need to convert the tuple into `{...}` in some place. 

Any suggestion?  I'm pretty new to Django..

Reply:
@ddio I'd have to checkout the PR and play with it a bit but it looks like passing a `tuple` won't work while a list was http://initd.org/psycopg/docs/usage.html#adapt-list

I wonder if `psycopg2` provides some kind of hashable/immutable `Array` object we can pass in there instead.

Reply:
Thanks! Will see if there's safe way to solve it this week.